File: CostadelSol_BackEnd/src/main/java/com/store/app/service/HuespedService.java
package com.store.app.service;

import java.util.List;

import com.store.app.entity.Huesped;

public interface HuespedService {
	
	public abstract List<Huesped> listAll();
	public List<Huesped> findAllByNameOrDniOrTelefono(String name, String lastName, String dni, String telefono);
	
	public abstract  Huesped RegistrarHuesped(Huesped obj);
	public abstract void EliminarHuesped(String id);

}

File: CostadelSol-Frontend/src/app/services/huesped.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Huesped } from '../models/huesped.model';

const baseUrl = "http://localhost:8090/registrarHuesped"

@Injectable({
  providedIn: 'root'
})
export class HuespedService {

  constructor(private http: HttpClient) { }

  register(data: Huesped): Observable<any> {
    return this.http.post(baseUrl + "/registrar", data)
  }

  listAll(): Observable<any> {
    return this.http.get<Huesped[]>(baseUrl + "/listAll")
  }

  listForNameOrDniOrTelefono(name: string, dni: string, telefono: string): Observable<any> {
    const params = new HttpParams()
      .set("name", name)
      .set("lastName", name)
      .set("dni", dni)
      .set("telefono", telefono);
    return this.http.get<Huesped[]>(baseUrl + "/listHuespedForNameOrDniOrTelefono", { params })
  }

}
